pandas/io/html.py

```diff
@@ -616,8 +616,8 @@ def _build_xpath_expr(attrs) -> str:
     if "class_" in attrs:
         attrs["class"] = attrs.pop("class_")
 
-    s = [f"@{k}={repr(v)}" for k, v in attrs.items()]
-    return "[{expr}]".format(expr=" and ".join(s))
+    s = " and ".join([f"@{k}={repr(v)}" for k, v in attrs.items()])
```

Comment:
if you end up making another pass through this file, can you do this in two lines?  i.e. keep `s` as it is now and do the join in a separate step